clang/lib/Sema/SemaStmt.cpp

```diff
@@ -2539,6 +2539,12 @@ StmtResult Sema::ActOnCXXForRangeStmt(Scope *S, SourceLocation ForLoc,
     return StmtError();
   }
 
+  if (getLangOpts().CPlusPlus23) {
+    auto Entity = InitializedEntity::InitializeVariable(RangeVar);
```

Comment:
Please spell out the type instead of using `auto`. Also, should this be a const ref to avoid a copy?

Reply:
I use`InitializedEntity ` instead of auto, `InitializedEntity::InitializeVariable` returns a rvalue, so I haven't add a const ref.

Reply:
Ah `InitializedEntity` doesn't have a copy constructor so the move operation will be cheap. Thanks!